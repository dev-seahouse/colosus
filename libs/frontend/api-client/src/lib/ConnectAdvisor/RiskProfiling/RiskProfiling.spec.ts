import ConnectAdvisorRiskProfilingApi from './RiskProfiling';

describe('Risk Profiling', () => {
  const connectAdvisorRiskProfilingApi = new ConnectAdvisorRiskProfilingApi();

  describe('getRiskProfiles()', () => {
    it('should return a valid response', async () => {
      const res = await connectAdvisorRiskProfilingApi.getRiskProfiles();

      expect(res.data).toMatchSnapshot();
    });
  });

  describe('getRiskQuestionnaire()', () => {
    it('should return a valid response', async () => {
      const res = await connectAdvisorRiskProfilingApi.getRiskQuestionnaire();

      expect(res.data).toMatchSnapshot();
    });
  });
});
