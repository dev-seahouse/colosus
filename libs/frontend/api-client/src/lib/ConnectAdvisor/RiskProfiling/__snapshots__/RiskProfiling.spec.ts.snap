// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Risk Profiling > getRiskProfiles() > should return a valid response 1`] = `
[
  {
    "id": "1c02181a-623d-4b6a-85d2-b528567498a0",
    "lowerLimit": "1",
    "riskProfileDescription": "You don’t want to experience volatility in your portfolio.<br/>You don’t expect the value of your portfolio to go down at any point in time.<br/>You should also understand that expected returns are very low.",
    "riskProfileName": "Low Risk",
    "tenantId": "9f0165e8-fa92-4bbf-808e-9dc7b6956970",
    "upperLimit": "1",
  },
  {
    "id": "0237a6bd-f27a-4ebf-a4ed-f0e77c27bf23",
    "lowerLimit": "2",
    "riskProfileDescription": "You are OK with a bit of volatility in your portfolio.<br/>You expect the value of your portfolio to go down but just for a short period of time before it bounces back.<br/>You also understand that expected returns are low.",
    "riskProfileName": "Medium-Low Risk",
    "tenantId": "9f0165e8-fa92-4bbf-808e-9dc7b6956970",
    "upperLimit": "2",
  },
  {
    "id": "2e746f8e-4c0b-4602-9c3b-e63733a1c4d7",
    "lowerLimit": "3",
    "riskProfileDescription": "You are OK with some volatility in your portfolio.<br/>You expect the value of your portfolio to to go down but just for a moderate period of time before it bounces back.<br/>You also understand that expected returns are average.",
    "riskProfileName": "Medium Risk",
    "tenantId": "9f0165e8-fa92-4bbf-808e-9dc7b6956970",
    "upperLimit": "3",
  },
  {
    "id": "8056cdf9-a6ee-4768-94a6-8c1344819a9b",
    "lowerLimit": "4",
    "riskProfileDescription": "You are OK with volatility in your portfolio.<br/>You expect the value of your portfolio to to go down but just for a significant period of time but it should bounce back.<br/>You expect good returns in the mid to long term.",
    "riskProfileName": "Medium-High Risk",
    "tenantId": "9f0165e8-fa92-4bbf-808e-9dc7b6956970",
    "upperLimit": "4",
  },
  {
    "id": "1bec2a79-cf97-4e00-9cff-b27af5c9d242",
    "lowerLimit": "5",
    "riskProfileDescription": "You are OK with high volatility in your portfolio.<br/>You expect the value of your portfolio to to go down sharply at one point in time but you know that you will reap huge benefits if you are patient enough.<br/>You expect high returns in the long term.",
    "riskProfileName": "High Risk",
    "tenantId": "9f0165e8-fa92-4bbf-808e-9dc7b6956970",
    "upperLimit": "5",
  },
]
`;

exports[`Risk Profiling > getRiskQuestionnaire() > should return a valid response 1`] = `
{
  "Questionnaire": [
    {
      "Questions": [
        {
          "Answers": [],
          "additionalConfiguration": {
            "questionFloorFlag": true,
            "questionNormalisationFactor1": "1",
            "questionNormalisationFactor2": "5",
            "scoreRangeConfig": [
              {
                "label": "<10%",
                "lowerBound": 0,
                "score": 1,
                "upperBound": 9,
              },
              {
                "answer": "10-25%",
                "lowerBound": 10,
                "score": 2,
                "upperBound": 24,
              },
              {
                "answer": "25-35%",
                "lowerBound": 25,
                "score": 3,
                "upperBound": 34,
              },
              {
                "answer": "35-50%",
                "lowerBound": 35,
                "score": 4,
                "upperBound": 49,
              },
              {
                "answer": ">50%",
                "lowerBound": 50,
                "score": 5,
                "upperBound": 100,
              },
            ],
          },
          "id": "c0b51a4b-4f17-42c9-8e63-2fa78dc52dc3",
          "question": "How much do you roughly save out of your monthly income?",
          "questionFormat": "NUMERIC_ENTRY",
          "questionType": "INCOME",
          "questionWeight": "0.5",
          "sortKey": 1,
        },
        {
          "Answers": [
            {
              "additionalConfiguration": {},
              "answer": "<35%",
              "id": "b6659756-973e-4f2c-8163-a3991610b817",
              "score": "5",
              "sortKey": 5,
            },
            {
              "additionalConfiguration": {},
              "answer": ">85%",
              "id": "417e9e23-88b7-4563-98e5-4638dacf4154",
              "score": "1",
              "sortKey": 5,
            },
            {
              "additionalConfiguration": {},
              "answer": "70-85%",
              "id": "347f5afd-28a6-469a-838a-42443b026c35",
              "score": "2",
              "sortKey": 4,
            },
            {
              "additionalConfiguration": {},
              "answer": "50-70%",
              "id": "4050301e-070d-4608-a8f8-636c3114b577",
              "score": "3",
              "sortKey": 3,
            },
            {
              "additionalConfiguration": {},
              "answer": "35-50%",
              "id": "ae152c9d-9291-4dad-bac0-cce28427a14b",
              "score": "4",
              "sortKey": 1,
            },
          ],
          "additionalConfiguration": {
            "questionFloorFlag": true,
            "questionNormalisationFactor1": "1",
            "questionNormalisationFactor2": "5",
          },
          "id": "3b216868-890a-4c47-9566-f5b6c983bd71",
          "question": "How much of your liquid assets are you planning to invest here?",
          "questionFormat": "SINGLE_CHOICE",
          "questionType": "BALANCE_SHEET",
          "questionWeight": "0.5",
          "sortKey": 2,
        },
      ],
      "additionalConfiguration": {
        "questionRoundFlag": true,
      },
      "groupingName": "FINANCIAL_HEALTH",
      "groupingType": "RISK_CAPACITY",
      "groupingWeight": "0.4",
      "id": "7ca3ad84-fa07-4bbf-8e5e-d21f5347df47",
      "scoringRules": "MAX",
      "sortKey": 1,
    },
    {
      "Questions": [
        {
          "Answers": [],
          "additionalConfiguration": {},
          "id": "b8d89946-6f33-4b77-8ea5-e49484421890",
          "question": "When do you want to achieve your goal?",
          "questionFormat": "NUMERIC_ENTRY",
          "questionType": "GOAL_TIME_FRAME",
          "questionWeight": "0",
          "sortKey": 1,
        },
      ],
      "additionalConfiguration": {
        "questionNormalisationFactor1": "1",
        "questionNormalisationFactor2": "12",
        "questionRoundDownFlag": true,
      },
      "groupingName": "GOAL",
      "groupingType": "RISK_CAPACITY",
      "groupingWeight": "0.2",
      "id": "422e5300-286f-4294-aa4b-b7adcc273bb2",
      "scoringRules": "MAX",
      "sortKey": 2,
    },
    {
      "Questions": [
        {
          "Answers": [],
          "additionalConfiguration": {},
          "id": "a624ebe3-ae95-4581-97d7-4edcae4c3f77",
          "question": "How old are you?",
          "questionFormat": "NUMERIC_ENTRY",
          "questionType": "AGE",
          "questionWeight": "0",
          "sortKey": 1,
        },
      ],
      "additionalConfiguration": {
        "questionNormalisationFactor1": "70",
        "questionNormalisationFactor2": "18",
        "questionRoundUpFlag": true,
      },
      "groupingName": "AGE",
      "groupingType": "RISK_CAPACITY",
      "groupingWeight": "0.2",
      "id": "cb5bcc9e-a911-4be0-834b-50c9e00902f8",
      "scoringRules": "MAX",
      "sortKey": 3,
    },
    {
      "Questions": [
        {
          "Answers": [
            {
              "additionalConfiguration": {},
              "answer": "Not familiar at all",
              "id": "1fd21a14-0bb5-40e6-9564-4178fb5461b8",
              "score": "1",
              "sortKey": 1,
            },
            {
              "additionalConfiguration": {},
              "answer": "Moderately familiar",
              "id": "c43a21fb-b0bf-4ff3-921d-8cc0f26cd93c",
              "score": "3",
              "sortKey": 3,
            },
            {
              "additionalConfiguration": {},
              "answer": "Somewhat familiar",
              "id": "c8f5da66-f60a-4a8d-a2f4-527addb2007f",
              "score": "2",
              "sortKey": 2,
            },
            {
              "additionalConfiguration": {},
              "answer": "Very familiar",
              "id": "d9ac8294-d708-4fda-8de2-dfbae938d3a9",
              "score": "4",
              "sortKey": 4,
            },
            {
              "additionalConfiguration": {},
              "answer": "Extremely familiar",
              "id": "51820ac7-a20f-496d-9be0-ade5d5e3cdf3",
              "score": "5",
              "sortKey": 5,
            },
          ],
          "additionalConfiguration": {},
          "id": "d5fe0921-a78f-42d7-8305-bf6014e0325e",
          "question": "How familiar are you with investing?",
          "questionFormat": "SINGLE_CHOICE",
          "questionType": "FINANCIAL_KNOWLEDGE",
          "questionWeight": "0",
          "sortKey": 1,
        },
      ],
      "additionalConfiguration": {
        "questionNormalisationFactor1": "1",
        "questionNormalisationFactor2": "5",
      },
      "groupingName": "FINANCIAL_KNOWLEDGE",
      "groupingType": "RISK_CAPACITY",
      "groupingWeight": "0.2",
      "id": "18e3b245-8138-4e01-85b8-0dd71c549e1e",
      "scoringRules": "MAX",
      "sortKey": 4,
    },
    {
      "Questions": [
        {
          "Answers": [
            {
              "additionalConfiguration": {},
              "answer": "High risk: this portfolio's value may fluctuate widely but I trust that it will pay off very significantly at one point in the future.",
              "id": "54808119-6022-4061-8315-84e1409d3435",
              "score": "4",
              "sortKey": 4,
            },
            {
              "additionalConfiguration": {},
              "answer": "No risk <br/>this portfolio's value should be slowly but surely going up.",
              "id": "93d246b3-c60a-47ba-92c0-4e965e13e3e1",
              "score": "1",
              "sortKey": 4,
            },
            {
              "additionalConfiguration": {},
              "answer": "Limited risk <br/>this portfolio's value can go down for a short period of time but its value should be going up most of the time.",
              "id": "c35126f0-9526-43da-bab7-83d9bad6546d",
              "score": "2",
              "sortKey": 3,
            },
            {
              "additionalConfiguration": {},
              "answer": "Significant risk <br/>this portfolio's value will go down for a certain period of time but it should recover if I am patient enough.",
              "id": "42109835-a9b8-4d49-97d0-bbed07f4a916",
              "score": "3",
              "sortKey": 2,
            },
          ],
          "additionalConfiguration": {
            "questionNormalisationFactor1": "1",
            "questionNormalisationFactor2": "4",
          },
          "id": "c22a1a56-ed1b-47c6-bda6-461c125ac0d4",
          "question": "What level of risk are you comfortable with for this portfolio?",
          "questionFormat": "SINGLE_CHOICE",
          "questionType": "RISK_COMFORT_LEVEL",
          "questionWeight": "0.5",
          "sortKey": 1,
        },
        {
          "Answers": [
            {
              "additionalConfiguration": {},
              "answer": "I will sell a part of this portfolio to minimise exposure, but keep some in hope that prices bounce back.",
              "id": "e92ee43e-776b-44eb-83e9-c4edc43bf9cf",
              "score": "2",
              "sortKey": 2,
            },
            {
              "additionalConfiguration": {},
              "answer": "I will invest more into this portfolio since its price is likely to bounce back.",
              "id": "525edb8d-02d5-4703-b60a-94e1b0241dea",
              "score": "4",
              "sortKey": 4,
            },
            {
              "additionalConfiguration": {},
              "answer": "I will continue holding onto my portfolio and hope it returns to the original price.",
              "id": "84caa058-0c20-403b-899a-236973aca339",
              "score": "3",
              "sortKey": 3,
            },
            {
              "additionalConfiguration": {},
              "answer": "I will fully redeem this portfolio to avoid further losses.",
              "id": "84171849-70a5-4325-ae6e-e8203f62c5b3",
              "score": "1",
              "sortKey": 1,
            },
          ],
          "additionalConfiguration": {
            "questionNormalisationFactor1": "1",
            "questionNormalisationFactor2": "4",
          },
          "id": "a5b99ad4-3e58-40ad-a114-5589d54c8b85",
          "question": "Imagine you witness a sudden drop in the value of this portfolio due to market fluctuations. What will you do?",
          "questionFormat": "SINGLE_CHOICE",
          "questionType": "RISK_COMFORT_LEVEL",
          "questionWeight": "0.5",
          "sortKey": 1,
        },
      ],
      "additionalConfiguration": {
        "questionCapFlag": true,
        "questionRoundDownFlag": true,
      },
      "groupingName": "RISK_COMFORT_LEVEL",
      "groupingType": "RISK_TOLERANCE",
      "groupingWeight": "0.5",
      "id": "8bed6ec9-b81f-4190-8ba7-d4f2ed4aea97",
      "scoringRules": "MAX",
      "sortKey": 1,
    },
  ],
  "activeVersion": 1,
  "id": "cbacafa5-7d7e-4d39-9bc3-6ce1839e3266",
  "questionnaireName": "Initial Questionnaire",
  "questionnaireType": "RISK_PROFILING",
}
`;
